azkaban-common/src/main/java/azkaban/jobtype/JobTypeManager.java

```diff
@@ -373,14 +393,158 @@ public Job buildJobExecutor(final String jobId, Props jobProps, final Logger log
+  }
 
+  /**
+   * Create an instance of Job with the given parameters, job id and job logger.
+   */
+  public static Job createJob(final String jobId, final JobParams jobParams, final Logger logger) {
+    Job job;
+    try {
+      job =
+          (Job) Utils.callConstructor(jobParams.jobClass, jobId, jobParams.pluginLoadProps,
+              jobParams.jobProps, jobParams.pluginPrivateProps, logger);
+    } catch (final Exception e) {
+      logger.info("Failed with 5 inputs with exception e = "
+          + e.getMessage());
+      job =
+          (Job) Utils.callConstructor(jobParams.jobClass, jobId, jobParams.pluginLoadProps,
+              jobParams.jobProps, logger);
+    }
     return job;
   }
 
+  public static final class JobParams {
+
+    public final Class<? extends Object> jobClass;
+    public final ClassLoader jobClassLoader;
+    public final Props jobProps;
+    public final Props pluginLoadProps;
+    public final Props pluginPrivateProps;
+
+    public JobParams(final Class<? extends Object> jobClass, final Props jobProps,
+                     final Props pluginPrivateProps, final Props pluginLoadProps) {
+      this.jobClass = jobClass;
+      this.jobClassLoader = jobClass.getClassLoader();
+      this.jobProps = jobProps;
+      this.pluginLoadProps = pluginLoadProps;
+      this.pluginPrivateProps = pluginPrivateProps;
+    }
+  }
+
   /**
    * Public for test reasons. Will need to move tests to the same package
    */
   public synchronized JobTypePluginSet getJobTypePluginSet() {
     return this.pluginSet;
   }
+
+  /**
+   * Expose cluster-specific libraries and native libraries through job properties.
+   * if a router is configured, construct the properties based on cluster.properties
+   * otherwise, the cluster is implicitly configured, the properties will be based
+   * on plugins' private properties.
+   */
+  private Props getClusterSpecificJobProps(final Cluster cluster, final Props jobProps,
+      final Props pluginProps) {
+    final Props clusterProps = new Props();
+
+    Props sourceProps;
+
+    if (cluster != null && !Cluster.UNKNOWN.equals(cluster)){
+      sourceProps = cluster.properties;
+      clusterProps.putAll(sourceProps);
+    } else {
+      // fall back to the existing mechanism if no cluster is found/configured
+      sourceProps = pluginProps;
+    }
+
+    final Collection<String> components = getClusterComponents(jobProps, pluginProps, true);
+
+    final String javaLibPath = Cluster.getJavaLibraryPath(sourceProps, components);
+    if (javaLibPath != null && !javaLibPath.isEmpty()) {
+      clusterProps.put(CommonJobProperties.TARGET_CLUSTER_CLASSPATH, javaLibPath);
+    }
+    final String nativeLibPath = Cluster.getNativeLibraryPath(sourceProps, components);
+    if (nativeLibPath != null && !nativeLibPath.isEmpty()) {
+      clusterProps.put(CommonJobProperties.TARGET_CLUSTER_NATIVE_LIB, nativeLibPath);
+    }
+    final String hadoopSecurityManagerClazz =
```

Comment:
Can you please rename it Clazz to class?

Reply:
addressed